File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        low = min(p.val,q.val)
        high = max(p.val,q.val)

        while True:
            if root.val > low and root.val > high:
                root = root.left
                continue
            elif root.val < low and root.val < low:
                root = root.right
                continue

            return root

File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {

        int max_vol = 0;

        int l = 0, r = heights.size() - 1;

        while (l < r) {

            int cur_vol = (r - l) * min(heights[l], heights[r]);

            max_vol = max(max_vol, cur_vol);

            if (heights[l] < heights[r]) 
                ++l;
            else
                --r;
        }

        return max_vol;
        
    }
};
